internal/ingress/controller/parser/parser_test.go

```diff
@@ -3105,6 +3105,56 @@ func TestGetEndpoints(t *testing.T) {
 				},
 			},
 		},
+		{
+			"should return one endpoint when the name of the port name match a port in the endpoint Subsets and" +
+				" service ipFamily is IPv6",
+			&corev1.Service{
+				Spec: corev1.ServiceSpec{
+					IPFamily:  getIPFamilyPtr(corev1.IPv6Protocol),
+					Type:      corev1.ServiceTypeClusterIP,
+					ClusterIP: "1::1",
+					Ports: []corev1.ServicePort{
+						{
+							Name:       "default",
+							TargetPort: intstr.FromInt(80),
+						},
+					},
+				},
+			},
+			&corev1.ServicePort{
+				Name:       "default",
+				TargetPort: intstr.FromInt(80),
+			},
+			corev1.ProtocolTCP,
+			func(string, string) (*corev1.Endpoints, error) {
+				nodeName := "dummy"
+				return &corev1.Endpoints{
+					Subsets: []corev1.EndpointSubset{
+						{
+							Addresses: []corev1.EndpointAddress{
+								{
+									IP:       "[1::1]",
```

Comment:
I'm a little confused here.

Is the following line of code not taking effect in this test case?

```go
epAddressIP = fmt.Sprintf("[%v]", epAddress.IP)
```

I expect the endpoint address to be of the form `1::1` and then it gets translated into `[1::1]` post processing.

What am I missing here?

Reply:
Thanks for your prompt. I have updated the unittest.